src/solvers/dgmulti/shock_capturing.jl

```diff
@@ -0,0 +1,284 @@
+# by default, return an empty tuple for volume integral caches
+function create_cache(mesh::DGMultiMesh{NDIMS}, equations,
+                      volume_integral::VolumeIntegralShockCapturingHG,
+                      dg::DGMultiFluxDiff{<:GaussSBP}, RealT, uEltype) where {NDIMS}
+  element_ids_dg   = Int[]
+  element_ids_dgfv = Int[]
+
+  # build element to element (EToE) connectivity for smoothing of
+  # shock capturing parameters.
+  FToF = mesh.md.FToF # num_faces x num_elements matrix
+  EToE = similar(FToF)
+  for e in axes(FToF, 2)
+    for f in axes(FToF, 1)
+      neighbor_face_index = FToF[f, e]
+
+      # reverse-engineer element index from face. Assumes all elements
+      # have the same number of faces.
+      neighbor_element_index = ((neighbor_face_index - 1) ÷ dg.basis.num_faces) + 1
+      EToE[f, e] = neighbor_element_index
+    end
+  end
+
+  # create sparse hybridized operators for low order scheme
+  Qrst, E = StartUpDG.sparse_low_order_SBP_operators(dg.basis)
+  Brst = map(n -> Diagonal(n .* dg.basis.wf), dg.basis.nrstJ)
+  sparse_hybridized_SBP_operators = map((Q, B) -> 0.5 * [Q-Q' E'*B; -B*E zeros(size(B))], Qrst, Brst)
+
+  # Find the joint sparsity pattern of the entire matrix. We store the sparsity pattern as
+  # a transpose for faster iteration through the rows.
+  sparsity_pattern = transpose(sum(map(A -> abs.(A), sparse_hybridized_SBP_operators)) .> 100 * eps())
```

Comment:
We used `'` (i.e., `adjoint`) for the flux differencing matrices, didn't we? I think it would be good to use the same types consistently.